rest/blip_sync.go

```diff
@@ -877,17 +876,25 @@ func (bh *blipHandler) handleRev(rq *blip.Message) error {
 		return base.HTTPErrorf(http.StatusBadRequest, "Missing docID or revID")
 	}
 
-	newDoc := &db.Document{
-		ID:    docID,
-		RevID: revID,
-	}
-	newDoc.UpdateBodyBytes(bodyBytes)
+	hasAttachments := bytes.Contains(bodyBytes, []byte(db.BodyAttachments))
+	hasExpiry := bytes.Contains(bodyBytes, []byte(db.BodyExpiry))
+
+	var body db.Body
+	var newDoc *db.IncomingDocument
+
+	newDoc = db.NewIncomingDocument(docID, bodyBytes)
 
 	if deltaSrcRevID, isDelta := revMessage.deltaSrc(); isDelta {
 		if !bh.sgCanUseDeltas {
 			return base.HTTPErrorf(http.StatusBadRequest, "Deltas are disabled for this peer")
 		}
 
+		// Unmarshal delta including properties for patching
+		err = rq.ReadJSONBody(&body)
```

Comment:
Is there a reason for doing this so much earlier than where body is used?

We have the opportunity to exit early below when deltaSrcRev isn't found.